xlators/features/trash/src/trash.c

```diff
@@ -1974,16 +1975,17 @@ trash_truncate(call_frame_t *frame, xlator_t *this, loc_t *loc, off_t offset,
         goto out;
     }
     /* This will be more accurate */
-    inode_path(loc->inode, NULL, &pathbuf);
+    retval = inode_path(loc->inode, NULL, &pathbuf);
```

Comment:
I think the best approach is to just add a `NULL` check for `pathbuf` after this function call.

Reply:
Similarly, in other places as well

Reply:
pathbuf is remaining NULL due to invalid inode so after returning EINVAL it  directly goto out rather than remaining function